### ISSUE TITLE
1.13 Bugs

### ISSUE THREAD

**Issue by JamesMilnerUK** [3drepo/3drepo.io#394]
## Reproducible

**James**

First Round of Testing
- [x] Log out not working - Logout in the Profile menu > Login in page > reload and it opens last opened account or “You are already logged in” message
- [x] Redirected to Login page in the middle of sign up 
- [x] IDs shown instead of names in Federation assignment
- [x] Team pop up shows ID instead of the model name
- [x] Opening and closing a model shows the ID instead of the name
- [x] Can’t create federation - dialog
- [x] Empty federations should not have a timestamp
- [x] Screenshot not formatted correctly when creating issue, OK once reloaded
- [x] Issue screenshots repeat themselves to cover the container
- [x] Show understandable message in deleting model in federation 
- [x] Team option missing from Federation menu - can’t share federation
- [x] Share option missing from Federation menu - can’t share federation
- [x] Projects folder is unnecessary 
- [x] No indication that accounts are loading
- [x] Login issue, occasionally logs you out unexpectedly after you've just logged in and then when you try to log back in it says 'already logged in'
- [x] Cookies etc force refresh the page
- [x] Units and model name should be required
- [x] Model still visible after logout
- [x] Issue shared by URL allows reposition pin / multiple pins
- [x] Sharing an Issue URL should take you to the location of the issue
- [x] Revisions, Settings and Download button doesn't work
- [x]  empty fed - time-stamp appears after reload
- [x]  navigation buttons - collapse when click in 3D view
- [x]  Federation editing should provide user feedback

Second Round of Testing
- [x] URL trailing slashes cause page switching to break [#397]
- [x] Going to unknown models just gives an empty view [#396]
- [x] Cannot create issue when there is no issue in the model
- [x] Issue URL doesn't go to the viewpoint anymore
- [x] New issue doesn't save and Pin icon is missing
- [x] Opening Default folder for any teamspace opens them all
- [x] Uploads are failing due to JavaScript error
- [x] No way to go back to project listing [#424]
- [x] Refreshing a model page gives an error and redirects me to teamspace
- [x] Priority Indication is missing
- [x] Model Properties is not applied onto the model
- [x] Cannot toggle off pin drop mode
- [x] Cannot remove pin
- [x] Clicking on the description go back to the viewpoint when you've clicked on a comment viewpoint
- [x] Going into model permission assignment page via the model doesn't seems to work
- [x] Sometimes add issue is disabled when it shouldn't be
- [x] Issues sizing has weird behaviour - not correct size all the time
- [x] Multiple error alerts on unity model crash
- [x] Selected objects are not saved
- [x] Licenses page shows no licenses even though I have some
- [x] Assign License is listed twice in Teamspace permission
- [x] Tree error (see below)
- [x] Going back model viewer to model list sometimes throws notification twice
- [x] State handling is borked
- [x] Pins sometimes bork: Could not find pin with name: 55e8cb80-6e23-11e7-8efd-5b2263c9675e
- [x] url unchanged after verification
- [x] Different pin colour depending on how you're opening the issue
- [x] Federation saving when the queue is busy
- [x] Permission assign page says Awaiting Models/Feds even though there are some
- [x] Going into model permission assignment page via the model still no worky
- [x] Tabs not clearing up
- [x] Going to signUp page whilst logged in causes it to go heywire
- [x] We need a way to revoke permissions 
- [x] Invoice CSS is borked
- [x] Clicking on invoices + history sometimes doesn't load correctly
- [ ] As an admin of the project I can't grant permissions on models within the project
- [x] I can see the edit/delete project menu even though I dont' have permissions to do anything
- [x] Allow user to know they are deleting child models if they delete a project
- [x] Importing a file seems to be pegged against the action of expanding its project folder
- [x] Translate backend error messages to user understandable front end error messages
- [x] Revisions take a while to load
- [x] Fix confirms to use the new dialog
- [x] Refreshing on ?page=modelsettings causes strange behaviour
- [x] settings button doesn't appear on a freshly created project:
- [x] Model settings doesn't load
- [x] automatically show someone who's a project admin/ teamspace admin to be an admin of the model on the permission list
- [x] Changing permissions on the model page doesn't work
- [ ] If you refresh whilst uploading a model, the front end doesn't hook back up with the chat server for progress update

Other things that aren't bugs:
- [x] Yarn
- [x] README Update
- [ ] README Screenshot

*Not Fixing*
- [ ] There's no way to change permissions on models in default project (Favour [#431])
- [ ] fed team - "No user matching "" was found" - can't add user to team (We are removing the Team UI)
- [ ] Federation's don't appear until the queue is unblocked (Favour #398 and #438)

**Henry**
- [x] Can’t close Issue 
- [x] Session error traceback sometimes returns
- [x] new error code from bouncer
- [x] default jobs
- [x] Chat server - doesn’t work both ways  
- [x] travis always error
- [x] show inherited/implied permissions on teamspace level
- [x] session-file-store library is very unstable on dev and the json session store on filesystem is sometimes corrupted, which is the culprit of the logout problem on dev => upgraded to 1.0.0
https://github.com/valery-barysok/session-file-store/issues/31
- [x] replace model id with name in a tree of a fed model

**Tim**

- [x] Navigation buttons - switching mode doesn’t show icons

**Carmen**
#424

**Comment by carmenfan** [comment]
Can we remove the "Project" folder while we're at it? It's redundant and it's just an extra click before we get to the list of projects

![image](https://user-images.githubusercontent.com/11945337/27581919-083dcbd8-5b27-11e7-962a-61446730ff01.png)

**Comment by thenrikie** [comment]
![image](https://user-images.githubusercontent.com/6147805/27732126-d0fc2436-5d87-11e7-99b9-82c325c46b02.png)
These buttons Revision, Settings and Download are not working because they use model name instead of model id when requesting APIs

**Comment by paloknapo** [comment]
- [ ] empty fed -  time-stamp appears after reload 
- [ ] fed team - "No user matching "" was found" - can't add user to team
- [ ] navigation buttons - collapse when click in 3D view

**Comment by carmenfan** [comment]
**Cannot create issue when there is no issue in the model**
For a model with no issues :
![image](https://user-images.githubusercontent.com/11945337/28320543-c50d5a74-6bc8-11e7-904b-0367d2ec8a54.png)
When you click on the + button to add a new issue, this happens:
![image](https://user-images.githubusercontent.com/11945337/28320563-d085a23a-6bc8-11e7-9221-5ed729a4cb2b.png)

**Comment by carmenfan** [comment]
**Issue URL doesn't go to the viewpoint anymore**
Entering a project with issue in the URL opens the issue but does not go to the viewpoint
![image](https://user-images.githubusercontent.com/11945337/28320869-9eb36656-6bc9-11e7-9b56-c5387f65a6c5.png)

**Comment by carmenfan** [comment]
**New issue doesn't save**
with an error "viewpoint is not defined" 
![image](https://user-images.githubusercontent.com/11945337/28320938-d2a281fe-6bc9-11e7-8560-8c9317a51fc5.png)

**Comment by carmenfan** [comment]
Not that important but I can see the "loading viewer" text on the project listing page :)

**Comment by carmenfan** [comment]
**Opening Default folder for any teamspace opens them all*(
- Expand a Default folder from any teamspace
![image](https://user-images.githubusercontent.com/11945337/28329924-945aae8e-6be3-11e7-81e7-e801ee2442e1.png)
- Expand another teamspace, and you can see the Default folder is expanded:
![image](https://user-images.githubusercontent.com/11945337/28329975-bda20346-6be3-11e7-986f-e5e37c7aaf4a.png)

This also applies for sub folders of Default folder. Which makes the scrolling go crazy

**Comment by carmenfan** [comment]
**Licenses page shows no licenses even though I have some**
I can see 3 licenses on my account in the database, but the licenses page is telling me I have no license, which is impossible as I have 2 members in my teamspace.
![image](https://user-images.githubusercontent.com/11945337/28372610-17627c62-6c98-11e7-8061-86ca08eda7b8.png)

subscription API also shows i have 3 licenses:
````js
[{"_id":"579b30134d831e1ef8ea2081","plan":"THE-100-QUID-PLAN","assignedUser":"carmen","active":true,"createdAt":"2016-07-29T10:29:39.000Z","updatedAt":"2016-07-29T10:29:39.000Z","expiredAt":null,"limits":{"spaceLimit":10737418240,"collaboratorLimit":1}},
{"_id":"579b30134d831e1ef8ea2082","plan":"THE-100-QUID-PLAN","active":true,"createdAt":"2016-07-29T10:29:39.000Z","updatedAt":"2016-07-29T10:29:39.000Z","expiredAt":null,"assignedUser":"jamesm","limits":{"spaceLimit":10737418240,"collaboratorLimit":1}},
{"_id":"5880e169c19cc7d3395edb70","plan":"THE-100-QUID-PLAN","active":true,"createdAt":"2017-01-19T15:55:21.000Z","updatedAt":"2017-01-19T15:55:21.000Z","expiredAt":null,"assignedUser":"pavol","limits":{"spaceLimit":10737418240,"collaboratorLimit":1}}]
````

One thing to note is that these licenses are assigned on our side instead of done through paypal (Enterprise, promotional/discretional type of licenses). So they are not shown in the billing page (which is correct), but they should be visible on the licenses page. So my guess is it's taking the same count as Billing, which is incorrect.

@thenrikie  knows more about this, but to tell whether a license is paypal-ed or otherwise, it depends on whether `inCurrentAgreement` is set to true. discretionary/enterprise licenses doesn't have this field (or it's set to false)

to summarise:
- Billing page should show the number of active licenses where inCurrentAgreement = true
- Licenses page should list all active licenses.

**Comment by carmenfan** [comment]
**Priority Indication is missing**
There is a little circle by the name of the issue creator that indicates the priority/status of the issue:
![image](https://user-images.githubusercontent.com/11945337/28373780-157c13d8-6c9b-11e7-8b5c-db630b24495a.png)
This is missing on the issue list:
![image](https://user-images.githubusercontent.com/11945337/28373881-2e87cdf4-6c9b-11e7-9143-aef10636dfc5.png)
Should look like this (from production): 
![image](https://user-images.githubusercontent.com/11945337/28373962-4c3c0efa-6c9b-11e7-8471-9f470f4c99c9.png)

**Comment by carmenfan** [comment]
**Refreshing a model page gives an error and redirects me to teamspace**
Open a model, hit refresh and you'll get this:
![image](https://user-images.githubusercontent.com/11945337/28374446-ac0522ee-6c9c-11e7-9c14-a5aaefebcf10.png)
this seems to be caused by a 404 error on https://dev.3drepo.io/api/carmen/46e66d55-4e41-453e-a14a-814b3d96ccd.json , which is weird as there's no reason for it to try to retrieve a json file of the model (is the front end thinking that it's the account page somehow?)

**Comment by carmenfan** [comment]
**Cannot toggle off pin drop mode**
pin operator should be a toggle, so the user can turn it on -> drop pin -> turn it off (pin remains in place) and continue interacting with the model(e.g. to highlight objects or move the camera view) without moving the pin. 

**Cannot remove pin**
If the user changes their mind and decided they don't want the pin, the pin should disappear when you click on the background. This doesn't work in dev.

**Comment by carmenfan** [comment]
**Clicking on the description go back to the viewpoint when you've clicked on a comment viewpoint**
open an issue -> click to change to viewpoint of the comment, click back on the description again, doesn't change the viewpoint back. In fact, it seems like the viewpoint associated with the description is set to the comment's viewpoint....

**Comment by carmenfan** [comment]
** Model Properties is not applied onto the model **
The particular duplex model we've been using has IFC spaces, which are hidden by default. This is determined by the modelProperties api, which gives you a list of IDs that should be hidden by default:
![image](https://user-images.githubusercontent.com/11945337/28375606-d0d34260-6c9f-11e7-829d-df2305eb6cbc.png)
This is no longer applied onto the model as soon as the model is loaded, the model looks like this:
![image](https://user-images.githubusercontent.com/11945337/28375661-f07217d6-6c9f-11e7-8989-bedcc3e70870.png)

When it should look like this:
![image](https://user-images.githubusercontent.com/11945337/28375714-1ba8e7fe-6ca0-11e7-80e0-3eab35740ca7.png)

**Comment by carmenfan** [comment]
**Going into model permission assignment page via the model doesn't seems to work**
doing this:
![image](https://user-images.githubusercontent.com/11945337/28376331-c5f9a832-6ca1-11e7-9a59-b335c3e713f5.png)
Seems to hang forever:
![image](https://user-images.githubusercontent.com/11945337/28376319-c09e0e6e-6ca1-11e7-963d-cf3efef55145.png)
I can get to that page via `Assign Permissions` on the left menu however:
![image](https://user-images.githubusercontent.com/11945337/28376424-fa8866ba-6ca1-11e7-928c-187fe950fd90.png)

**Comment by carmenfan** [comment]
**Assign License is listed twice in Teamspace permission**
![image](https://user-images.githubusercontent.com/11945337/28376558-541d308e-6ca2-11e7-86d5-f3d2dbbabb6a.png)

**Comment by carmenfan** [comment]
**There's no way to change permissions on models in default project**
I know we don't want people to use the default project, but not letting them share any models inside is a bit mean... :laughing: 
![image](https://user-images.githubusercontent.com/11945337/28376665-9cc951aa-6ca2-11e7-8b87-46b5724daac3.png)

**Comment by JamesMilnerUK** [comment]
Note to self:

```
if(vm.idToPath[objectID]){
	path = vm.idToPath[objectID].split("__");
} else {
	path = vm.subProjIdToPath[objectID].split("__");
	var parentPath = vm.subTreesById[path[0]].parent.path.split("__");
	path = parentPath.concat(path);
}

Sometimes throws an error

**Comment by carmenfan** [comment]
**url unchanged after verification**
Sign up -> click on link from verification email

Once you're verified it redirects you to teamspace, but url remains unchanged:
![image](https://user-images.githubusercontent.com/11945337/28469628-be1f90c0-6e2e-11e7-8fce-5330ff6c8d2e.png)
which after a while it changes back to the verification message saying something along the lines of failed to verify.

**Comment by carmenfan** [comment]
**Different pin colour depending on how you're opening the issue**
Click on a pin gives you the older colour style: 
![image](https://user-images.githubusercontent.com/11945337/28469786-4ea636a8-6e2f-11e7-8867-7f7631d15e38.png)
If you make the pin select from the issue component you'll get a more yellowy colour:
![image](https://user-images.githubusercontent.com/11945337/28469809-618fc450-6e2f-11e7-8e37-1da8c1396625.png)

We should probably unify all this and just have one constant somewhere that depicts pin colour (potentially in the config!)

**Comment by carmenfan** [comment]
**Federation saving when the queue is busy**
This really should just behave like models when you click save:
![image](https://user-images.githubusercontent.com/11945337/28470267-221c1c18-6e31-11e7-8ca1-8c79e221dbbc.png)

This is blocking the user from doing anything while waiting.

I cancelled the federation processing (which btw doesn't cancel the queue request , so it is still going to get processed as long as you've clicked save), refreshed and the federation entry does not appear on my list, so I retried with the same federation name and this error appears:
![image](https://user-images.githubusercontent.com/11945337/28470464-ab8e322e-6e31-11e7-95dd-98a289c0ec4f.png)
(It should be federation already exists)

This is expected as the model entry is already written in the database, but what is not expected is that the entry isn't listed until the queue finished. This is not the same behaviour as the models, and there isn't a lot of reason to make it that way either.

(To replicate this, just try to upload a relatively large model before you try to create a federation)

**Comment by carmenfan** [comment]
**Permission assign page says Awaiting Models/Feds even though there are some**
Looked at permissions management page for models/feds of a project with models, looks fine:
![image](https://user-images.githubusercontent.com/11945337/28471013-89dfd978-6e33-11e7-8be0-ad049d807700.png)

Switch to a different project (which also have models), and it hangs on awaiting models/feds:
![image](https://user-images.githubusercontent.com/11945337/28471041-a5eca66e-6e33-11e7-9c73-c621712c19f5.png)

Switching back to Chair and I've lost my list of models as well!
![image](https://user-images.githubusercontent.com/11945337/28471060-b5093b94-6e33-11e7-8e43-7aba356345bb.png)

**Comment by carmenfan** [comment]
** Going into model permission assignment page via the model doesn't seems to work still no worky**
![image](https://user-images.githubusercontent.com/11945337/28471726-029f4310-6e36-11e7-9d76-a11a2e6eab17.png)

**Comment by carmenfan** [comment]
**Tabs not clearing up**
Sign up -> keep a browser window on the "Thank you fro signing up blah blah.."  (Let's call this window A)
Click verify from email (which will spawn up another 3D Repo window) -> log in (window B)
Look at the window A once logged in, it will redirect you to the teamspace page, but the verification tab still remains:
![image](https://user-images.githubusercontent.com/11945337/28472262-ce27b048-6e37-11e7-88c9-6961391cd943.png)

If I log out from Window A, I get back to that "thank you" blah blah:
![image](https://user-images.githubusercontent.com/11945337/28472284-e265a66e-6e37-11e7-871e-6bcd2a5c4c2f.png)

Actually this happens on any page that doesn't require sign in when you are already logged in:
![image](https://user-images.githubusercontent.com/11945337/28472310-006f2d74-6e38-11e7-8478-9d5f70ca76fe.png)
![image](https://user-images.githubusercontent.com/11945337/28472351-2474564a-6e38-11e7-9122-67bc3f12e905.png)

**Comment by carmenfan** [comment]
**Permission changes does not save**
I altered the permissions like this:
![image](https://user-images.githubusercontent.com/11945337/28474235-bbd9fd44-6e3f-11e7-9b97-a275621e612a.png)

Refreshed my page and went to the same screen, my changes are not applied:
![image](https://user-images.githubusercontent.com/11945337/28474260-d555089a-6e3f-11e7-9509-a804503fbe32.png)

I feel like we're missing a save button

**Comment by carmenfan** [comment]
> Just a thought looking at this: Should we have another radio button there that says "No Access"? As it stands it looks like there's no way to revoke the permission to view. Also a little weird how "pavol" here just doesn't have an option selected

We definitely need a way to revoke permissions before we release this.

**Comment by carmenfan** [comment]
As discussed just now:
- Deleting a project should have a warning that says something along the lines of "Are you sure? All your models will be gone!!!!"
- Deleting a project should remove all the models within the project instead of pushing them to default folder

**Comment by carmenfan** [comment]
**As an admin of the project I can't grant permissions on models within the project **
I'm a project admin of henry's teamspace and the UI doesn't let me alter the permissions within it

**Comment by carmenfan** [comment]
**I can see the edit/delete project menu even though I dont' have permissions to do anything**
Just a collaborator, but this appears:
![image](https://user-images.githubusercontent.com/11945337/28712138-e3710c84-7381-11e7-862b-27d316c11020.png)
We probably shouldn't even give the options to the user if they don't have the right permissions. 

But if you decided that this should stay, currently there's no error messages if you try to change the name of the project

**Comment by carmenfan** [comment]
**Importing a file seems to be pegged against the action of expanding its project folder**
* Create a model + upload a file while you're at it
* The upload/processing of the model file does not happen until you expand the project folder
* Once the upload completes/fails, if you collapse & expand the project folder again, another request will be launched

**Comment by carmenfan** [comment]
**Translate backend error messages to user understandable front end error messages**
At the moment, if the bouncer returns error code -1, which translates to bouncer failed to launch, displays the user with an error message that says "Bouncer failed to launch", which makes no sense to the user. We need to map across all the potential messages to something that is more user friendly

**Comment by carmenfan** [comment]
settings button doesn't appear on a freshly created project:
![image](https://user-images.githubusercontent.com/11945337/29071335-07bb533a-7c3b-11e7-8377-09670b6450af.png)
It will only appear again after refreshing:
![image](https://user-images.githubusercontent.com/11945337/29071372-2e1a02a6-7c3b-11e7-948b-59a9fb0450b7.png)

**Comment by carmenfan** [comment]
**Model settings doesn't load**
Try to open a model setting in Pavol's Teamspace:
![image](https://user-images.githubusercontent.com/11945337/29071448-9bfc4ebe-7c3b-11e7-8068-4669e83deff8.png)

Get a 404 not found because it's calling an api with carmen/<modelID> instead of pavol/<modelID>:
![image](https://user-images.githubusercontent.com/11945337/29071485-bc932102-7c3b-11e7-8457-e4e138da3c35.png)

**Comment by carmenfan** [comment]
We should automatically show someone who's a project admin/ teamspace admin to be an admin of the model on the permission list:

Carmen is admin of the teamspace but on model level permission she is listed as unassigned
![image](https://user-images.githubusercontent.com/11945337/29080078-c7b043e2-7c56-11e7-9b39-3bac428428a5.png)

**Comment by carmenfan** [comment]
Changing permissions on the model page doesn't actually work:
The permission list for this model looks like this:
![image](https://user-images.githubusercontent.com/11945337/29080280-4b49cf5c-7c57-11e7-9b5e-13436e6eaa62.png)

I change pavol to `unassigned`:
![image](https://user-images.githubusercontent.com/11945337/29080302-5940de20-7c57-11e7-823c-6061dfbcdb41.png)

refresh -> the permission list is unchanged:
![image](https://user-images.githubusercontent.com/11945337/29080342-71874a1e-7c57-11e7-86b2-9ccbf63b8368.png)

Looking at the network tab, when you change someone's permission, the frontend is doing a POST request with the old permissions, thus not actually changing any permissions at all:
![image](https://user-images.githubusercontent.com/11945337/29080420-9f56c7ee-7c57-11e7-972b-2c68c564c85c.png)

**Comment by carmenfan** [comment]
*If you refresh whilst uploading a model, the front end doesn't hook back up with the chat server for progress update*

* Upload a relatively large file
* wait till it's `processing`
* refresh the page
* `processing` is gone, but the model isn't ready

**Comment by carmenfan** [comment]
**Forgot password page is missing logo**
![image](https://user-images.githubusercontent.com/11945337/29215276-424f74b8-7ea2-11e7-85f1-038d9b2b12b2.png)

**Comment by carmenfan** [comment]
**Sign Up page the logo is obscured**
![image](https://user-images.githubusercontent.com/11945337/29215313-69b7a980-7ea2-11e7-9925-9c7c1e815f8c.png)

**Comment by carmenfan** [comment]
**Privacy, cookies, T&C pages are empty**
![image](https://user-images.githubusercontent.com/11945337/29215359-a8cf641e-7ea2-11e7-9057-4b01650117c4.png)

There's also duplicated pricing/contact links:
![image](https://user-images.githubusercontent.com/11945337/29215401-cbb8f5f8-7ea2-11e7-863b-f43f05aa53a7.png)
